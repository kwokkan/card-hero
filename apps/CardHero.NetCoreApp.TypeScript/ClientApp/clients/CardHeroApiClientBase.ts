// no export as this is used by NSwag generated api clients
class CardHeroApiClientBase {
    getBaseUrl(defaultBaseUrl: string, baseUrl: string): string {
        if (!baseUrl) {
            return defaultBaseUrl;
        }

        if (baseUrl.endsWith('/')) {
            // remove trailing slash as NSwag uses '/' as part of the api url
            return baseUrl.substring(0, baseUrl.length - 2);
        }

        return baseUrl;
    }
}

module.exports = CardHeroApiClientBase;