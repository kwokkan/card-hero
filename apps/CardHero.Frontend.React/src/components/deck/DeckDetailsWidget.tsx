import React from "react";
import { IDeckModel } from "../../clients/clients";
import { Icon } from "../../styles/index";

interface IDeckDetailsWidgetProps {
    deck: IDeckModel;
    onSaveClicked?: (deck: IDeckModel) => void;
}

export function DeckDetailsWidget(props: IDeckDetailsWidgetProps): JSX.Element {
    const deck = props.deck;

    const onSavedClicked = () => {
        if (props.onSaveClicked) {
            props.onSaveClicked(props.deck)
        }
    };

    return (
        <div className="card">
            <h4 className="card-header">
                {deck.name}
                <Icon icon="star" className={"deck-favourite float-right" + (deck.isFavourited ? " enabled" : "")} data-deck-id={deck.id} />
            </h4>
            <ul className="list-group list-group-flush">
                <li className="list-group-item">
                    <strong>Description:</strong>
                    {" "}
                    <span >{deck.description}</span>
                </li>

                <li className="list-group-item">
                    <strong>Max cards:</strong>
                    {" "}
                    <span>{deck.maxCards}</span>
                </li>
            </ul>
            <div className="card-footer">
                <button type="button" id="save-deck" className="btn btn-success pull-right" onClick={onSavedClicked} data-deck-id={deck.id}>Save</button>
            </div>
        </div>
    );
}
