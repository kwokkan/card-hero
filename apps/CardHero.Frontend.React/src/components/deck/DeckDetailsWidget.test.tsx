import React from "react";
import renderer from "react-test-renderer";
import { IDeckModel } from "../../clients/clients";
import { DeckDetailsWidget } from "./DeckDetailsWidget";

test("<DeckDetailsWidget /> renders correctly without favourite", () => {
    const deck: IDeckModel = {
        id: 1,
        maxCards: 5,
        name: "Test deck"
    };
    const tree = renderer
        .create(
            <DeckDetailsWidget deck={deck} />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<DeckDetailsWidget /> renders correctly with favourite", () => {
    const deck: IDeckModel = {
        id: 1,
        isFavourited: true,
        maxCards: 5,
        name: "Test deck"
    };
    const tree = renderer
        .create(
            <DeckDetailsWidget deck={deck} />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<DeckDetailsWidget /> renders correctly with description", () => {
    const deck: IDeckModel = {
        description: "Test deck description",
        id: 1,
        maxCards: 5,
        name: "Test deck"
    };
    const tree = renderer
        .create(
            <DeckDetailsWidget deck={deck} />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
