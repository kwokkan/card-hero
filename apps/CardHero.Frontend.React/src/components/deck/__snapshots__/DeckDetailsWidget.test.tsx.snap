// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeckDetailsWidget /> renders correctly with description 1`] = `
<div
  className="card"
>
  <h4
    className="card-header"
  >
    Test deck
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-star fa-w-18 fa-fw deck-favourite float-right"
      data-deck-id={1}
      data-icon="star"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </h4>
  <ul
    className="list-group list-group-flush"
  >
    <li
      className="list-group-item"
    >
      <strong>
        Description:
      </strong>
       
      <span>
        Test deck description
      </span>
    </li>
    <li
      className="list-group-item"
    >
      <strong>
        Max cards:
      </strong>
       
      <span>
        5
      </span>
    </li>
  </ul>
  <div
    className="card-footer"
  >
    <button
      className="btn btn-success pull-right"
      data-deck-id={1}
      id="save-deck"
      onClick={[Function]}
      type="button"
    >
      Save
    </button>
  </div>
</div>
`;

exports[`<DeckDetailsWidget /> renders correctly with favourite 1`] = `
<div
  className="card"
>
  <h4
    className="card-header"
  >
    Test deck
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-star fa-w-18 fa-fw deck-favourite float-right enabled"
      data-deck-id={1}
      data-icon="star"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </h4>
  <ul
    className="list-group list-group-flush"
  >
    <li
      className="list-group-item"
    >
      <strong>
        Description:
      </strong>
       
      <span />
    </li>
    <li
      className="list-group-item"
    >
      <strong>
        Max cards:
      </strong>
       
      <span>
        5
      </span>
    </li>
  </ul>
  <div
    className="card-footer"
  >
    <button
      className="btn btn-success pull-right"
      data-deck-id={1}
      id="save-deck"
      onClick={[Function]}
      type="button"
    >
      Save
    </button>
  </div>
</div>
`;

exports[`<DeckDetailsWidget /> renders correctly without favourite 1`] = `
<div
  className="card"
>
  <h4
    className="card-header"
  >
    Test deck
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-star fa-w-18 fa-fw deck-favourite float-right"
      data-deck-id={1}
      data-icon="star"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </h4>
  <ul
    className="list-group list-group-flush"
  >
    <li
      className="list-group-item"
    >
      <strong>
        Description:
      </strong>
       
      <span />
    </li>
    <li
      className="list-group-item"
    >
      <strong>
        Max cards:
      </strong>
       
      <span>
        5
      </span>
    </li>
  </ul>
  <div
    className="card-footer"
  >
    <button
      className="btn btn-success pull-right"
      data-deck-id={1}
      id="save-deck"
      onClick={[Function]}
      type="button"
    >
      Save
    </button>
  </div>
</div>
`;
